.github/renovate.json5

```diff
@@ -0,0 +1,46 @@
+{
+  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
+  "dependencyDashboard": true,
+  "suppressNotifications": ["prEditedNotification"],
+  "extends": ["config:base"],
+  "labels": ["internal"],
+  "schedule": ["before 4am on Monday"],
+  "separateMajorMinor": false,
+  "enabledManagers": [
+    "github-actions",
+    "pre-commit",
+    "cargo",
+    "pep621",
+    "npm",
+  ],
+  "pep621": {
+    "fileMatch": [
+      "^(python|scripts)/.*pyproject\\.toml$",
+    ],
+  },
+  "npm": {
+    "fileMatch": [
+      "^playground/.*package\\.json$",
+    ],
+  },
+  "pre-commit": {
+    "enabled": true,
+  },
+  "ignoreDeps": [
+    // The latest versions of these are not compatible with our release workflow
+    "actions/upload-artifact",
+    "actions/download-artifact",
+  ],
+  "packageRules": [
+    {
+      "groupName": "GitHub Actions",
+      "matchManagers": ["github-actions"],
+      "description": "Weekly update of GitHub Action dependencies",
```

Comment:
Can we only match the upload/download artifact group? e.g.

https://github.com/astral-sh/uv/blob/main/.github/dependabot.yml#L9-L12

Reply:
Yes! Sorry, I missed that.

Reply:
It's different over here — good opportunity to fix.

Reply:
Oh, so right now:
- Ruff:
  - doesn't do any updates for `actions/upload-artifact` and `actions/download-artifact`: https://github.com/astral-sh/ruff/blob/f7aab5ac69f9505a1776a4bcf20ea2415aa5e442/.github/dependabot.yml#L12-L15
  - all other GitHub Actions updates are bundled into a single weekly PR: https://github.com/astral-sh/ruff/blob/f7aab5ac69f9505a1776a4bcf20ea2415aa5e442/.github/dependabot.yml#L8-L11
- uv:
  - bundles `actions/upload-artifact` and `actions/artifact` updates into a single weekly PR: https://github.com/astral-sh/uv/blob/7a5571fa5c4fa72a26b0754f3273f1d75f3fa7d2/.github/dependabot.yml#L9-L12
  - all other GitHub Actions updates are done in individual PRs on a weekly basis

What do we want the new behaviour for Ruff to be?

Reply:
I forgot to remove the actions from the dependabot configuration after doing the major version upgrade. I think upgrading them together (when both versions are bumped) does make sense, which is what UV does

Reply:
Do we want:
1. All actions updates bundled together in one weekly PR?
2. The upload/download actions bundled into one weekly PR, but an individual weekly PR each for all other actions updates?
3. The upload/download actions bundled into one weekly PR, and all other actions updates bundled into a second weekly PR?

Reply:
2.